internal/uuid/uuid_test.go

```diff
@@ -13,44 +13,56 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
-func TestNew(t *testing.T) {
-	m := make(map[UUID]bool)
-	for i := 1; i < 1000000; i++ {
-		uuid, err := New()
-		require.NoError(t, err, "New error")
-		require.False(t, m[uuid], "New returned a duplicate UUID %v", uuid)
-		m[uuid] = true
-	}
-}
-
 // GODRIVER-2349
 // Test that initializing many package-global UUID sources concurrently never leads to any duplicate
 // UUIDs being generated.
 func TestGlobalSource(t *testing.T) {
-	// Create a slice of 1,000 sources and initialize them in 1,000 separate goroutines. The goal is
-	// to emulate many separate Go driver processes starting at the same time and initializing the
-	// uuid package at the same time.
-	sources := make([]*source, 1000)
-	var wg sync.WaitGroup
-	for i := range sources {
-		wg.Add(1)
-		go func(i int) {
-			defer wg.Done()
-			sources[i] = newGlobalSource()
-		}(i)
-	}
-	wg.Wait()
+	t.Run("exp rand 1 UUID x 1,000,000 goroutines", func(t *testing.T) {
+		// Read a UUID from each of 1,000,000 goroutines and assert that there is never a duplicate value.
+		const iterations = 1e6
+		uuids := new(sync.Map)
+		var wg sync.WaitGroup
+		for i := 1; i < iterations; i++ {
+			wg.Add(1)
```

Comment:
```suggestion
		var wg sync.WaitGroup
		wg.Add(iterations)
		for i := 1; i < iterations; i++ {
```

This seems a bit cleaner, but totally optional.

Reply:
It's a weird thing I don't quite understand yet. You can see I use `wg.Add(iterations)` in the other test case, where `iterations` is 1000. But the same code here does not exit as expected when the `iterations` is 1e6. To be honest, it's my first time to pass a value other than 1 to `wg.Add`.

Reply:
@qingyang-hu Oh okay, no worries!